quickwit/quickwit-common/src/test_utils.rs

```diff
@@ -31,11 +31,11 @@ pub async fn wait_until_predicate<Fut>(
     retry_interval: Duration,
 ) -> Result<(), Elapsed>
 where
-    Fut: Future<Output = Result<bool, anyhow::Error>>,
+    Fut: Future<Output = bool>,
 {
     tokio::time::timeout(timeout, async move {
         loop {
-            if predicate().await.is_ok() {
+            if predicate().await {
```

Comment:
Forgot to fix it in my previous PR, we were not testing the right condition before...